src/containers/PageContainer.jsx

```diff
@@ -46,6 +57,16 @@ class PageContainer extends React.PureComponent {
     return contents;
   }
 
+  getQuestionsWithNumbers = (questionNumbers, questions) => {
+    if (!questions || !questionNumbers) return [];
+
+    questions.forEach((question, index) => {
+      question.number = questionNumbers[index];
+    });
+
+    return questions;
+  };
+
```

Comment:
basically what you had, just using it in a bit more straight forward way now that  we have this discreet object where we have arrays of question numbers organized by page id.